### ISSUE TITLE
discovery_target_decode is considered a component rather than standard lib function

### ISSUE THREAD

**Issue by mariuspodean** [grafana/agent#4305]
### What's wrong?

The following error is raised when using teh std lib function `discovery_target_decode`:

`Error: /etc/agent-config/agent.river:27:14: component "discovery_target_decode" does not exist `

It seems that discovery_target_decode is considered a component rather a standard lib function. I don't have similar issues with the rest of the functions from the std lib.
I'm running version v0.34.3 of the agent in docker.

### Steps to reproduce

The bug can be reproduced by running the example code found in the documentation: https://grafana.com/docs/agent/latest/flow/reference/stdlib/discovery_target_decode/

```
local.file "example" {
  filename = env("TARGETS_FILE")
}

prometheus.scrape "default" {
  targets = discovery_target_decode(local.file.example.content)
}
```

### System information

Docker

### Software version

v0.34.3

**Comment by rfratto** [comment]
Related: #3607

**Comment by djcode** [comment]
I see that there's discussion in removing this function and using better methods outlined in #3607 but until the time those better methods are baked in, this function should keep working so this appears to be a regression.

I did a bit more digging into this as I also need this function for something.
This function broke between v0.33.2 and v0.34.0. It seems to be specifically the commit f4082af876055065e7190899a99e2d6d9b8cca70
The function logic was moved from `pkg/river/internal/stdlib/stdlib.go` to a new file `pkg/flow/internal/stdlib/stdlib.go`.
I think this new file just needs to be imported somethere.
